app/services/search_query_sanitization_service.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+class SearchQuerySanitizationService
+  ANYTHING_EXCEPT_NUMBERS_AND_SPACE_REGEX = /[^[[:digit:]] ]+/.freeze
+  ANYTHING_EXCEPT_NUMBERS_SPACE_PERIOD_REGEX = /[^[[:digit:]]\. ]+/.freeze
+  ANYTHING_EXCEPT_NUMBERS_REGEX = /[^[[:digit:]]]+/.freeze
+  ANYTHING_EXCEPT_LETTERS_NUMBERS_HYPHEN_SPACE_REGEX = /[^[[:alnum:]] -]+/.freeze
+
+  MAX_QUERY_TERM_LENGTH = 256 # characters
+
+  def initialize; end
+
+  ##
+  # @param [ActionController::Parameters] params
+  # @return [Hash]
+  #
+  def sanitize_for_standard_search(params)
+    return {} if params.nil?
+
+    clean_search_term          = sanitize_search_term(params['s'])
+    clean_handshape            = sanitize_handshape(params['hs'])
+    clean_body_location_fields = sanitize_body_location_fields(params['l'])
+    clean_body_location_groups = sanitize_body_location_groups(params['lg'])
+    clean_usage                = sanitize_usage(params['usage'])
+    clean_tag                  = sanitize_tag(params['tag'])
+
+    result = {}
+
+    result['s']     = [clean_search_term]                   unless clean_search_term.empty?
+    result['hs']    = clean_handshape.split(' ')            unless clean_handshape.empty?
+    result['l']     = clean_body_location_fields.split(' ') unless clean_body_location_fields.empty?
+    result['lg']    = clean_body_location_groups.split(' ') unless clean_body_location_groups.empty?
+    result['usage'] = clean_usage.split(' ')                unless clean_usage.empty?
+    result['tag']   = clean_tag.split(' ')                  unless clean_tag.empty?
+
+    result
+  end
+
+  def sanitize_for_autocomplete_search(term)
+    sanitize_search_term(term)
+  end
+
+  private
+
+  ##
+  # @param [String] term
+  # @return [String]
+  #
+  def sanitize_search_term(term)
+    return '' if term.nil?
+
+    term
+      .gsub(ANYTHING_EXCEPT_LETTERS_NUMBERS_HYPHEN_SPACE_REGEX, '')
+      .strip
+      .truncate(MAX_QUERY_TERM_LENGTH, omission: '')
+  end
+
+  ##
+  # @param [String] handshape
+  # @return [String]
+  #
+  def sanitize_handshape(handshape)
+    return '' if handshape.nil?
+
+    handshape
+      .gsub(ANYTHING_EXCEPT_NUMBERS_SPACE_PERIOD_REGEX, '')
```

Comment:
the long names did make me smile, but they are good names

Reply:
(but btw in NZ we call that a fullstop, not a period, afaik)

Reply:
Same in Ireland but that was even longer! 😀